certora/harnesses/GuardianStorageHarness.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.12 <0.9.0;
+import {GuardianStorage} from "../../contracts/modules/social_recovery/storage/GuardianStorage.sol";
+
+contract GuardianStorageHarness is GuardianStorage {
+
+    /**
+     * @notice Gets the count of guaridans for a wallet based on the actual linked list.
+     * @dev This is needed for FV to avoid cases where we have values which does not start with SENTINEL.
+     * @param _wallet The target wallet.
+     * @return count of guardians.
+     */
+    function countGuardians(address _wallet) public view returns (uint256 count) {
+        GuardianStorageEntry storage entry = entries[_wallet];
+        if (entry.count == 0 || entry.guardians[SENTINEL_GUARDIANS] == SENTINEL_GUARDIANS || entry.guardians[SENTINEL_GUARDIANS] == address(0)) {
+            return 0;
+        }
+
+        address currentGuardian = entry.guardians[SENTINEL_GUARDIANS];
+        while (currentGuardian != SENTINEL_GUARDIANS && currentGuardian != address(0)) {
+            currentGuardian = entry.guardians[currentGuardian];
+            count++;
+        }
```

Comment:
```suggestion
        // The sentinel guardian pointing to address 0 is the initial state for the
        // guardian storage entry for an account and is equivalent to an empty list
        // where the sentinel points to itself. We handle this special case here.
        if (entry.guardians[SENTINEL_GUARDIANS] == address(0)) {
            return 0;
        }

        address currentGuardian = entry.guardians[SENTINEL_GUARDIANS];
        while (currentGuardian != SENTINEL_GUARDIANS) {
            currentGuardian = entry.guardians[currentGuardian];
            count++;
        }
```

Reply:
The `entry.guardians[SENTINEL_GUARDIANS] == SENTINEL_GUARDIANS` is already handled by the `while` loop condition and doesn't need to be special cased and a guardian entry should never point to 0.

At some point, we need to prove the integrity of this harness function, but for that we need invariants for the guard storage. Can you create a GH issue to track this please?

Reply:
Updated 👍🏾

Reply:
The `while` loop was still not updated.

Reply:
Now it made sense to me why the check was moved inside the `while` 👍🏾 

Regarding the issue, it was created already: #24